### PR TITLE
Add task space control

### PR DESCRIPTION
**Pull request by hjzh4**
- Add task space control for ros/gazebo.

### CODE REVIEW

File: garage/contrib/ros/envs/sawyer/reacher_env.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def __init__(self,
                         delta range the goal is randomized
         :param robot_control_mode: string
                         robot control mode: 'position' or 'velocity'
-                        or 'effort'
+                        or 'effort' or 'gripper_position'
```
Comment (zhanpenghe) [inline garage/contrib/ros/envs/sawyer/reacher_env.py]: Please use task_space_control

File: garage/contrib/ros/robots/sawyer.py [hunk 2]
```diff
@@ -140,9 +151,12 @@ def get_observation(self):
             list(self._limb.joint_efforts().values()))
 
         obs = np.concatenate(
-            (gripper_pos, gripper_ori, gripper_lvel, gripper_avel,
-             gripper_force, gripper_torque, robot_joint_angles,
+            (endpoint_pos, endpoint_ori, endpoint_lvel, endpoint_avel,
```
Comment (zhanpenghe) [inline garage/contrib/ros/robots/sawyer.py]: To match the mujoco environment we have, for torque/effort control we used joint_angles, joint_vel and end effector position as observation space.

**Comment by hjzh4** [comment]
Considering README.md, I'm feeling I am helping users to set up the workstation for sawyer. However, users are expected to finish set up before they want to run any experiments on it using garage.contrib.ros. I think it would be too tedious to help users with everything, I am planning to leave only links to intera tutorials. Do you have any ideas about it.

File: garage/contrib/ros/robots/sawyer.py (hunk 2 above)
Comment (zhanpenghe) [reply garage/contrib/ros/robots/sawyer.py]: Yes. But at least for environments like reacher we don't need that many fields as observation(e.g. endpoint_ori). I think a good way to solve this is have a get_full_observation for the sawyer robot class and customizes the environment observations.

**Comment by zhanpenghe** [comment]
Yes, but we want to at least have a successful run after following the guidance (on real robot and simulation).

File: garage/contrib/ros/envs/sawyer/README.md [hunk 3]
```diff
@@ -137,37 +137,41 @@ $ catkin_make -DPYTHON_EXECUTABLE:FILEPATH=$GARAGE_PYTHON
 ```
 ### Setup MoveIt! for safety check
 #### Install MoveIt!
+**You need flexible collision library with fcl-0.5 version**
```
Comment (ryanjulian) [inline garage/contrib/ros/envs/sawyer/README.md]: Does this provide install instructions including fcl-0.5?

File: garage/contrib/ros/envs/sawyer/README.md [hunk 4]
```diff
@@ -18,8 +18,8 @@ $ export GARAGE=/path/to/your/garage/
 ```
 ## Installation
 - Assuming we got a clean system which only has garage running.
-- Check if our robots and workstations are supported by garage.contrib.ros
-- Following is how we setup the garage.contrib.ros's environment for sawyer robot
+- Check if our robots and workstations are supported by garage.garage.contrib.ros
+- Following is how we setup the garage.garage.contrib.ros's environment for sawyer robot
```
Comment (ryanjulian) [inline garage/contrib/ros/envs/sawyer/README.md]: I think you mean garage.contrib.ros

File: garage/contrib/ros/envs/sawyer/README.md [hunk 5]
```diff
@@ -205,7 +209,15 @@ $ source activate garage
...
 ## Troubleshooting
 ### ImportError: import cv2 when running training script
 Make sure your are using opencv-python installed in garage conda env.
+### Need to use boost_python-py35
+    - See https://github.com/osmcode/pyosmium/issues/52
+    - We need boost_python-py35
+    ```bash
+    # quick and dirty workaround
+    rm /usr/lib/x86_64-linux-gnu/libboost_python.so
+    ln -s /usr/lib/x86_64-linux-gnu/libboost_python-py35.so /usr/lib/x86_64-linux-gnu/libboost_python.so
```
Comment (ryanjulian) [inline garage/contrib/ros/envs/sawyer/README.md]: The latest conda environment is based on python36. Will this work?

File: garage/contrib/ros/envs/sawyer/launchers/trpo_real_sawyer_push.py [hunk 6]
```diff
@@ -22,22 +28,26 @@
 def run_task(*_):
     initial_goal = np.array([0.6, -0.1, 0.80])
 
+    # Initialize moveit_commander
+    moveit_commander.roscpp_initialize(sys.argv)
```
Comment (ryanjulian) [inline garage/contrib/ros/envs/sawyer/launchers/trpo_real_sawyer_push.py]: Can this be done from inside the SawyerEnv instead?

File: garage/contrib/ros/robots/sawyer.py [hunk 7]
```diff
@@ -17,7 +18,9 @@ class Sawyer(Robot):
     def __init__(self,
                  initial_joint_pos,
                  moveit_group,
-                 control_mode='position'):
+                 without_gripper=False,
```
Comment (ryanjulian) [inline garage/contrib/ros/robots/sawyer.py]: Please make boolean names always affirmative, i.e. `has_gripper=True`

File: garage/contrib/ros/robots/sawyer.py (hunk 2 above)
Comment (ryanjulian) [reply garage/contrib/ros/robots/sawyer.py]: This interface doesn't have any use if it doesn't match the simulation environment. Please make it match. If the mujoco environment needs to have things/options added, please do that. They are a matching pair.

File: garage/contrib/ros/envs/sawyer/README.md (hunk 5 above)
Comment (hjzh4) [reply garage/contrib/ros/envs/sawyer/README.md]: Yea, it works so far.

File: garage/contrib/ros/envs/sawyer/launchers/trpo_real_sawyer_push.py (hunk 6 above)
Comment (hjzh4) [reply garage/contrib/ros/envs/sawyer/launchers/trpo_real_sawyer_push.py]: it needs to be called before rospy.init_node, we put ropy.init_node in launcher file for some reasons before. I will take a look later.

**Comment by ryanjulian** [comment]
closing because this stale. please reopen when ready.

**Pull request closed without merging**